### PR TITLE
Show server tasks in terminal

### PR DESCRIPTION
**Pull request by akaroml**
Signed-off-by: Rome Li <rome.li@microsoft.com>

Depends on: eclipse/eclipse.jdt.ls#1280

### CODE REVIEW

File: src/serverTasks.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	return newArray;
+}
+
+function applyReport(report: ProgressReport) {
+	const index = findIndex(tasks, task => task.id === report.id);
+	if (index === -1) {
+		tasks.push(report);
+		organizeTasks();
+		return;
+	}
```
[the remark this answers is not in the document]
Comment (akaroml) [reply src/serverTasks.ts]: applied, thanks!

File: src/commands.ts [hunk 2]
```diff
@@ -168,4 +168,8 @@ export namespace Commands {
      * Navigate To Super Method Command.
      */
     export const NAVIGATE_TO_SUPER_IMPLEMENTATION_COMMAND = 'java.action.navigateToSuperImplementation';
+    /**
+     * Show server task status
+     */
+    export const SHOW_SERVER_TASK_STATUS = 'java.show.server.stask.status';
```
[the remark this answers is not in the document]
Comment (akaroml) [reply src/commands.ts]: fixed. thanks!

File: src/serverTasks.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+function organizeTasks() {
+	let newArray = tasks;
+	if (tasks.length > suggestedTaskEntrySize) {
+		newArray = recycleTasks(tasks, suggestedTaskEntrySize);
+	}
+
+	// make sure in-progress items are always at the end
+	newArray = newArray.filter(task => task.complete);
+	newArray.push(...tasks.filter(task => !task.complete));
```
[the remark this answers is not in the document]
Comment (akaroml) [reply src/serverTasks.ts]: Modified. Thanks!

**Comment by akaroml** [comment]
ping... @fbricon

**Comment by fbricon** [comment]
Nice job @akaroml !

**Comment by jclingan** [comment]
This feature creates a new terminal window by default. As a Java developer, why should I care about this? It is providing me all kinds of info that is not relevant to the project I am working on.  If I set  `java.progressReports.enabled=false` to stop the window creation, the terminal is still created but with no content. How can I prevent this terminal window when working on Java projects?

**Comment by jclingan** [comment]
When I "Add a file to workspace" for the first time, an untitled workspace is created and vscode tries to create a second instance of the server status window. This generates the attached error dialog.

Java_Overview_—_Untitled__Workspace_

**Comment by akaroml** [comment]
@jclingan ack. I can reproduce this issue. Will improve the support for multi-root workspace. Thanks for reporting.

**Comment by jclingan** [comment]
Thanks @akaroml !

**Pull request merged**